src/key_io.cpp

```diff
@@ -309,3 +311,33 @@ bool IsValidDestinationString(const std::string& str)
 {
     return IsValidDestinationString(str, Params());
 }
+
+std::pair<CPubKey, CPubKey> DecodeSilentData(const std::vector<unsigned char>& data)
+{
+
+    std::vector<unsigned char> scan_pubkey_data(data.begin(), data.begin() + 33);
+    CPubKey scan_pubkey{scan_pubkey_data};
+
+    std::vector<unsigned char> spend_pubkey_data(data.begin() + 33, data.end());
+    CPubKey spend_pubkey{spend_pubkey_data};
+
+    return {scan_pubkey, spend_pubkey};
+}
+
+std::vector<unsigned char> DecodeSilentAddress(const std::string& str)
```

Comment:
Similar to "Decode", there should be a higher-level function that first decodes the string and then passes it to the relevant decoding function

Reply:
Even better, should probably first add a new destination type to CTxDestination

Reply:
Resolved in #28246 , which adds a new `CTxDestination` and lets us use the existing encoder and decoder.